java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -102,12 +96,12 @@ public static long serialize(WriteChannel out, Schema schema) throws IOException
   /**
    * Deserializes a schema object. Format is from serialize().
    *
-   * @param in the channel to deserialize from
+   * @param reader the reader interface to deserialize from
    * @return the deserialized object
    * @throws IOException if something went wrong
    */
-  public static Schema deserializeSchema(ReadChannel in) throws IOException {
-    Message message = deserializeMessage(in);
+  public static Schema deserializeSchema(MessageReader reader) throws IOException {
```

Comment:
This method seems to closer to `read schema` rather than `deserialize schema`

```
public static Schema deserializeSchema(Message message)
```
seem to make more sense to me

Maybe this method can be made into:
```
public static Schema readSchema(MessageReader reader) {

    Message message = reader.readNextMessage();
    return deserializeSchema(message);
}
```
?

@BryanCutler what do you think

Reply:
I think it's ok to include reading the message as part of deserialization and some messages also require to read another chunk after the message.  I do think the behavior of these functions could be made to be more consistent, but we should probably do that as a followup.

Reply:
Ok. Agree this can be a follow upl